OXRS-SHA-StateIO-ESP32-FW.ino

```diff
@@ -834,11 +843,13 @@ void scanI2CBus()
     if (Wire.endTransmission() == 0)
     {
       bitWrite(g_mcps_found, mcp, 1);
```

Comment:
If we added `g_mcps_found = 0;` at the start of `scanI2CBus()` then I think this could be dynamically re-run if/when the `ioConfig` setting is changed?

Or we split `scanI2CBus()` into two methods - one which does the scanning and populating `g_mcps_found`, and then a second which is called after `rack32.begin()` and initialises the MCP pins and input/output handlers.

So then we only scan the bus once, on startup, but can then initialise the MCPs whenever the `ioConfig` changes.

Reply:
Yes, I think splitting into two methods as you suggest should do it.